src/main/java/com/google/flourbot/datastorage/FirebaseDataStorage.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.flourbot.datastorage;
+
+import com.google.api.core.ApiFuture;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.firestore.DocumentReference;
+
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.FirestoreOptions;
+
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.Query;
+import com.google.cloud.firestore.QueryDocumentSnapshot;
+import com.google.cloud.firestore.QuerySnapshot;
+import com.google.cloud.firestore.WriteResult;
+import com.google.common.collect.ImmutableMap;
+
+import java.io.*;
+
+public class FirebaseDataStorage implements DataStorage {
+    private Firestore db;
+
+    public FirebaseDataStorage () {
+        try {
+            this.db = initializeFirebase();
+        }
+        catch (FileNotFoundException e) {
+            System.out.println("Service key not found.");
+        }
+        catch (Exception e) {
+            System.out.println("Firebase not initialized.");
+        }
+    }
+ 
+    private Firestore initializeFirebase() throws Exception {
+        FileInputStream serviceAccount;     
+        serviceAccount = new FileInputStream("key.json");
```

Comment:
Do I need to close this input stream?

Reply:
@SydneyCaulfeild  All the examples I've seen for initializing service accounts have not closed the stream so I think you're good!

Reply:
Taken from Effective Java, `try-with-resources close` FileInputStream automatically.
```// try-with-resources - the the best way to close resources!
static String firstLineOfFile(String path) throws IOException {
    try (BufferedReader br = new BufferedReader(
        new FileReader(path))) {
            return br.readLine();
       }
} 
```

Reply:
Ok thanks!